cmd/geth/main.go

```diff
@@ -445,9 +444,11 @@ func startNode(ctx *cli.Context, stack *node.Node) {
 			}
 		}()
 	}
+
+	isFullNode := ctx.GlobalString(utils.SyncModeFlag.Name) == "full" || ctx.GlobalString(utils.SyncModeFlag.Name) == "fast"
```

Comment:
Seems like `&&` go turned into `||`

Reply:
yeah, actually this is correct, but the if below as `!isFullNode`